# Глава 20 – Исключения

___

## Вопрос 1

Напишите класс дроби Fraction с конструктором, который принимает числитель и знаменатель. Если пользователь передает знаменатель 0, генерируется исключение типа std::runtime_error (включено в заголовок stdexcept). В функции main() попросите пользователя ввести два целых числа. Если дробь будет корректна, выведите ее. Если дробь некорректна, перехватите std::exception и сообщите пользователю, что он ввел недопустимую дробь.

Вот что должна вывести программа при выполнении:

```cpp
Enter the numerator: 5
Enter the denominator: 0
Invalid denominator

```

__Ответ:__

[Решение](Вопрос_1/Вопрос_1.cpp)


```

